controllers/staff.js

```diff
@@ -112,6 +133,31 @@ const getAllOrders = async (rid) => {
     return await db.any(psGetAllOrders, [rid]);
 }
 
+const getRestaurantPromos = async (rid) => {
+    return await db.any(psGetRestaurantPromos, [rid]);
+}
+
+const insertRestaurantPromos = async (rid, item) => {
+    const pid = await db.one(psInsertPromo, [item.startdate, item.enddate, item.points, item.percentoff, item.minspending, item.monthswithnoorders])
+        .then((res) => {
+            db.none(psInsertRestaurantPromo, [rid, res.pid]);
```

Comment:
Might be better if can write as one transaction, in case insertRestaurantPromo fails, it can rollback the insertpromo.